marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Server.hs

```diff
@@ -307,24 +315,28 @@ execCreate contractQueryConnector getCurrentScripts solveConstraints loadWalletC
       , payoutScriptHash = payoutScript
       }
   let
-    contractSafetyErrors =
-      if False  -- FIXME: Disabled because of incompatibility with integration tests.
-        then checkContract networkId roleTokens version contract' continuations
-        else mempty
-  -- FIXME: The is a placeholder until we design safety-analysis reporting.
-  unless (null contractSafetyErrors)
-    . throwE . CreateSafetyAnalysisError
-    $ show contractSafetyErrors
+    -- Fast analysis of safety: examines bounds for transactions.
+    contractSafetyErrors = checkContract networkId roleTokens version contract' continuations
+    limitAnalysisTime =
+      liftIO
+        . fmap (either Right id)
+        . race ([SafetyAnalysisTimeout] <$ threadDelay (floor $ nominalDiffTimeToSeconds analysisTimeout * 1_000_000))
```

Comment:
There is no time limit on the fast analysis of safety, because that doesn't use `z3` to find all paths through the contract and it doesn't execute any Plutus transactions.